File: AndroidJava/AndroidFragmentAPP/app/src/main/java/dev/donmanuel/androidfragmentapp/MainActivity.java
package dev.donmanuel.androidfragmentapp;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import dev.donmanuel.androidfragmentapp.Fragment.FirstFragment;
import dev.donmanuel.androidfragmentapp.Fragment.SecondFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.main, new FirstFragment())
                    .commit();
        }

    }

    public void showSecondFragment() {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.main, new SecondFragment())
                .addToBackStack(null)
                .commit();
    }
}

File: FirestoreCleanArchitectureApp/app/src/main/java/dev/donmanuel/firestoreclean/presentation/MainActivity.kt
package dev.donmanuel.firestoreclean.presentation

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import dagger.hilt.android.AndroidEntryPoint
import dev.donmanuel.firestoreclean.presentation.book_list.BookListScreen

@AndroidEntryPoint
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            BookListScreen()
        }
    }
}